### PR TITLE
Update Github templates.

### PR DESCRIPTION
**Pull request by 1uc**
Addresses #830.

### CODE REVIEW

**Review by alkino — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/build_failure.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+1. Please paste text as text and not as a screen shot.
+2. If in doubt paste too much output rather than to little, i.e. don't be too
+   scared of a large wall of text. Especially, if it's a compiler error.
+   (Anything past the first error is largely uninformative and can be safely
+   stripped.)
+3. Please strip all boilerplate.
+
+Markdown summary:
+````
+Some include `code` and a block:
```
Comment (1uc) [inline .github/ISSUE_TEMPLATE/build_failure.md:47]:
```suggestion
Some inline `code` and a block:
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -8,24 +8,31 @@ assignees: ''
 ---
 
 **Describe the bug**
-A clear and concise description of what the bug is.
+Please give a clear and concise description of the bug. For us to be able to
+fix the bug we need to be able to reproduce it. As a result, if you provide a
+reproducer, i.e. code exhibits the bug (incl. build instructions) but is
```
Comment (1uc) [inline .github/ISSUE_TEMPLATE/bug_report.md:13]:
```suggestion
reproducer, i.e. code that exhibits the bug (incl. build instructions) but is
```

File: .github/ISSUE_TEMPLATE/build_failure.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+**Bug Description**
+Unfortunately, build failures tend to be highly specific and there's many
+things that could go wrong on both our and your side. Therefore, a reproducer
+is essential. You could try the minimal setup found here:
+https://github.com/BlueBrain/HighFive/blob/master/doc/installation.md#manually-install-highfive
+
+from there you can work upwards by adding complexity until you reproduce the
+issue.
+
+Once you have a reproducer, please paste the exact `cmake` command used to
```
Comment (1uc) [inline .github/ISSUE_TEMPLATE/build_failure.md:19]:
```suggestion
Once you have a reproducer, please paste it and the exact `cmake` command used to
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -8,24 +8,31 @@ assignees: ''
...
+  - Compiler:
+  - [optional] OS:
+  - [optional] CMake:
+  - [optional] HDF5:
 
-**Expected behavior**
-A clear and concise description of what you expected to happen.
+**Style Guide**
+1. Please paste text as text and not as a screen shot.
+2. If in doubt paste too much output rather than to little, i.e. don't be too
```
Comment (heerener) [inline .github/ISSUE_TEMPLATE/bug_report.md:26]: `s/to little/too little/`

File: .github/ISSUE_TEMPLATE/build_failure.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+**Version Information**
+  - HighFive:
+  - Compiler:
+  - OS:
+  - CMake:
+  - HDF5:
+
+**Style Guide**
+1. Please paste text as text and not as a screen shot.
+2. If in doubt paste too much output rather than to little, i.e. don't be too
```
Comment (1uc) [inline .github/ISSUE_TEMPLATE/build_failure.md:39]:
```suggestion
2. If in doubt paste too much output rather than too little, i.e. don't be too
```

**Review by heerener — APPROVED** [review]

**Pull request merged**